asset-transfer-basic/chaincode-go/assetTransfer.go

```diff
@@ -0,0 +1,179 @@
+
+		if err != nil {
+			return fmt.Errorf("Failed to put to world state. %s", err.Error())
+		}
+	}
+
+	return nil
+}
+
+// CreateAsset issues a new asset to the world state with given details.
+func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface, assetID, id, color, size, owner, appraisedValue string) error {
+	asset := Asset{
+		ID:             id,
+		Color:          color,
+		Size:           size,
+		Owner:          owner,
+		AppraisedValue: appraisedValue,
+	}
+
+	assetAsBytes, _ := json.Marshal(asset)
+
+	return ctx.GetStub().PutState(assetID, assetAsBytes)
+}
+
+// ReadAsset returns the asset stored in the world state with given id.
+func (s *SmartContract) ReadAsset(ctx contractapi.TransactionContextInterface, assetID string) (*Asset, error) {
+	assetAsBytes, err := ctx.GetStub().GetState(assetID)
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read from world state. %s", err.Error())
+	}
+	if assetAsBytes == nil {
+		return nil, fmt.Errorf("%s does not exist", assetID)
+	}
+
+	asset := new(Asset)
+	_ = json.Unmarshal(assetAsBytes, asset)
+
+	return asset, nil
+}
+
+// UpdateAsset updates an existing asset in the world state with provided parameters.
+func (s *SmartContract) UpdateAsset(ctx contractapi.TransactionContextInterface, assetID, id, color, size, owner, appraisedValue string) error {
+	assetAsBytes, err := ctx.GetStub().GetState(assetID)
+
+	if err != nil {
+		return fmt.Errorf("Failed to read from world state. %s", err.Error())
+	}
+	if assetAsBytes == nil {
+		return fmt.Errorf("%s does not exist", assetID)
+	}
+
+	// overwritting original asset with new asset
+	return s.CreateAsset(ctx, assetID, id, color, size, owner, appraisedValue)
+}
+
+// DeleteAsset deletes an given asset from the world state.
+func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface, assetID string) error {
+	return ctx.GetStub().DelState(assetID)
+}
+
+// AssetExists checks is an given asset exists in the world state.
+func (s *SmartContract) AssetExists(ctx contractapi.TransactionContextInterface, assetID string) (bool, error) {
+	assetAsBytes, err := ctx.GetStub().GetState(assetID)
+
+	if err != nil {
+		return false, fmt.Errorf("Failed to read from world state. %s", err.Error())
+	}
+
+	return assetAsBytes != nil, nil
+}
+
+// TransferAsset updates the owner field of asset with given id in world state.
+func (s *SmartContract) TransferAsset(ctx contractapi.TransactionContextInterface, assetID string, newOwner string) error {
+	asset, err := s.ReadAsset(ctx, assetID)
+
+	if err != nil {
+		return err
+	}
+
+	asset.Owner = newOwner
+
+	assetAsBytes, _ := json.Marshal(asset)
+
+	return ctx.GetStub().PutState(assetID, assetAsBytes)
+}
+
+// GetAllAssets returns all assets found in world state
+func (s *SmartContract) GetAllAssets(ctx contractapi.TransactionContextInterface) ([]QueryResult, error) {
+	startKey := "ASSET0"
+	endKey := "ASSET99"
```

Comment:
startKey and endKey should be empty string for full range query (Fabcar got this wrong...)

Reply:
done.